dissect/target/plugins/apps/texteditor/windowsnotepad.py

```diff
@@ -0,0 +1,213 @@
+from __future__ import annotations
+
+import zlib
+from typing import Iterator
+
+from dissect.cstruct import cstruct
+
+from dissect.target.exceptions import UnsupportedPluginError
+from dissect.target.helpers.descriptor_extensions import UserRecordDescriptorExtension
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.target.helpers.record import (
+    UnixUserRecord,
+    WindowsUserRecord,
+    create_extended_descriptor,
+)
+from dissect.target.plugin import export
+from dissect.target.plugins.apps.texteditor.texteditor import (
+    GENERIC_TAB_CONTENTS_RECORD_FIELDS,
+    TexteditorPlugin,
+)
+
+# Thanks to @Nordgaren, @daddycocoaman, @JustArion and @ogmini for their suggestions and feedback in the PR
+# thread. This really helped to figure out the last missing bits and pieces
+# required for recovering text from these files.
+
+c_def = """
+struct header {
+    char        magic[2]; // NP
+    uint8       unk0;
+    uint8       fileState; // 0 if unsaved, 1 if saved
+}
+
+struct header_saved_tab {
+    uleb128     filePathLength;
+    wchar       filePath[filePathLength];
+    uleb128     fileSize;
+    uleb128     encoding;
+    uleb128     carriageReturnType;
+    uleb128     timestamp; // Windows Filetime format (not unix timestamp)
```

Comment:
From my and @JustArion testing, the timestamp is a uleb128. If we assume a uint64, it doesn't provide a valid Win32 filetime.  

![image](https://github.com/fox-it/dissect.target/assets/112595633/ef622824-192a-434e-a3ba-d8fa32322d74)

In the above example, we have A2 AC 8A 9E D3 A1 A3 ED 01 or 133574226803332642 in uleb128. This converts to 4/12/2024 7:11:20 PM in GMT. https://www.epochconverter.com/ldap 

This matches with the modified time on the original file. Which is 4/12/2024 3:11:20 PM in EST

![image](https://github.com/fox-it/dissect.target/assets/112595633/4ed77abe-d008-41d3-a5c9-574d193886ea)

This also lines up with the Filetime structure in the Win32 API (https://learn.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-filetime) which specifies a 64-bit value.

Reply:
Thanks for the clarification @ogmini <3! Im keeping the comment unresolved for now, because I believe this is perfect info that could be returned to the analist.

Reply:
Good suggestion! In the latest commit, I added a separate record that also presents the `saved_path`, `sha256` and `modification_time` fields to the analyst.